### DISCUSSION TITLE
jsonable_encoder() vs dict() to obtain a dict from a BaseModel object

### DISCUSSION THREAD

**Discussion by imyosh** [fastapi/fastapi#8878] [Questions] [answered]
Apparently there are two ways to convert a BaseModel object to a dict format :
the native BaseModel dict() function and the fastapi jsonable_encoder() that used internally by the framework .

which one should I use ?
is there a difference in the performance ?

**Comment by paloderosa** [comment, marked as answer]
Both produce dictionaries, but `dict` might produce objects like `datetime` or `Enum`, while `jsonable_encoder` produces serialized fields (it would convert those objects into `str`, for example).
